src/test/cdc/t/cdctestlib.pm

```diff
@@ -0,0 +1,351 @@
+    }
+
+    my $client_config_options = "
+max_connections = 100
+max_wal_senders = 100
+max_replication_slots = 100
+";
+    $node->init(allows_streaming => 'logical');
+    if ($node_type == $NODE_TYPE_COORDINATOR || $node_type == $NODE_TYPE_WORKER) {
+        $node->append_conf("postgresql.conf",$citus_config_options);
+    } else {
+        $node->append_conf("postgresql.conf",$citus_config_options);
+    }
+
+    $node->start();
+
+    if ($node_type == $NODE_TYPE_COORDINATOR || $node_type == $NODE_TYPE_WORKER) {
+        $node->safe_psql('postgres', "CREATE EXTENSION citus;");
+        my $value = $node->safe_psql('postgres', "SHOW citus.enable_change_data_capture;");
+        print("citus.enable_change_data_capture value is $value\n")
+    }
+
+    return $node;
+}
+
+# Create a Citus cluster with the given number of workers
+sub create_citus_cluster {
+    my ($no_workers,$host,$port,$citus_config) = @_;
+    my @workers = ();
+    my $node_coordinator;
+    print("citus_config :", $citus_config);
+    if ($citus_config ne "") {
+        $node_coordinator = create_node('coordinator', $NODE_TYPE_COORDINATOR,$host, $port, $citus_config);
+    } else {
+        $node_coordinator = create_node('coordinator', $NODE_TYPE_COORDINATOR,$host, $port);
+    }
+    my $coord_host = $node_coordinator->host();
+    my $coord_port = $node_coordinator->port();
+    $node_coordinator->safe_psql('postgres',"SELECT pg_catalog.citus_set_coordinator_host('$coord_host', $coord_port);");
+    for (my $i = 0; $i < $no_workers; $i++) {
+        $port = $port + 1;
+        my $node_worker;
+        if ($citus_config ne "") {
+            $node_worker = create_node("worker$i", $NODE_TYPE_WORKER,"localhost", $port, $citus_config);
+        } else {
+            $node_worker = create_node("worker$i", $NODE_TYPE_WORKER,"localhost", $port);
+        }
+        my $node_worker_host = $node_worker->host();
+        my $node_worker_port = $node_worker->port();
+        $node_coordinator->safe_psql('postgres',"SELECT pg_catalog.citus_add_node('$node_worker_host', $node_worker_port);");
+        push @workers, $node_worker;
+    }
+    return $node_coordinator, @workers;
+}
+
+sub create_cdc_publication_and_replication_slots_for_citus_cluster {
+    my $node_coordinator = $_[0];
+    my $workersref = $_[1];
+    my $table_names = $_[2];
+
+    create_cdc_publication_and_slots_for_coordinator($node_coordinator, $table_names);
+    create_cdc_publication_and_slots_for_workers($workersref, $table_names);
+}
+
+sub create_cdc_publication_and_slots_for_coordinator {
+    my $node_coordinator = $_[0];
+    my $table_names = $_[1];
+    print("node node_coordinator connstr: \n" . $node_coordinator->connstr());
+    $node_coordinator->safe_psql('postgres',"CREATE PUBLICATION cdc_publication FOR TABLE $table_names;");
+    $node_coordinator->safe_psql('postgres',"SELECT pg_catalog.pg_create_logical_replication_slot('cdc_replication_slot','citus',false,true)");
```

Comment:
why do we create pg_create_logical_replication_slot with twophase = true? Just curious

Reply:
Just used from an example in the documentation.
Changed the "two_phase" parameter to "false" now.